src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1138,6 +1135,83 @@ def add_units(
+    write_units_table_in_processing_module: bool = False,
+):
+    """
+    Primary method for writing a SortingExtractor object to an NWBFile.
+    Parameters
+    ----------
+    sorting: SpikeInterfaceSorting
+    save_path: PathType
+        Required if an nwbfile is not passed. The location where the NWBFile either exists, or will be written.
+    overwrite: bool
+        If using save_path, whether or not to overwrite the NWBFile if it already exists.
+    nwbfile: NWBFile
+        Required if a save_path is not specified. If passed, this function
+        will fill the relevant fields within the nwbfile. E.g., calling
+        spikeextractors.NwbRecordingExtractor.write_recording(
+            my_recording_extractor, my_nwbfile
+        )
+        will result in the appropriate changes to the my_nwbfile object.
+    skip_features: list of str
+        Each string in this list that matches a spike feature will not be written to the NWBFile.
+    units_table_name : str (optional, defaults to 'units')
+        The name of the units table. If write_as=='units', then units_table_name must also be 'units'.
+    unit_table_description : str (optional)
+        Text description of the units table; it is recommended to include information such as the sorting method,
+        curation steps, etc.
+    write_units_table_in_processing_module: bool (optional, defaults to False)
+        How to save the units table in the nwb file.
+        - True will save it to the processing module to serve as a historical provenance for the official table.
+        - False will save it to the official NWBFile.Units position; recommended only for the final form of the data.
+    """
+    if isinstance(sorting, SortingExtractor):
+        sampling_frequency = sorting.get_sampling_frequency()
+        if sampling_frequency is None:
+            raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
+        checked_sorting = OldToNewSorting(oldapi_sorting_extractor=sorting)
+    else:
+        checked_sorting = sorting
+    unit_ids = checked_sorting.get_unit_ids()
+    sampling_frequency = checked_sorting.get_sampling_frequency()
+
+    if write_units_table_in_processing_module:
+        ecephys_mod = get_module(
+            nwbfile=nwbfile,
+            name="ecephys",
+            description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP.",
+        )
+        if units_table_name not in ecephys_mod.data_interfaces:
+            units_table = pynwb.misc.Units(name=units_table_name, description=unit_table_description)
+            ecephys_mod.add(units_table)
+
+        units_table = ecephys_mod[units_table_name]
+    else:
+        if nwbfile.units is None:
+            nwbfile.units = pynwb.misc.Units(name=units_table_name, description=unit_table_description)
+        units_table = nwbfile.units
+
     if isinstance(sorting, SortingExtractor):
         all_features = set()
```

Comment:
Also can you add a commented placeholder section for the `else` condition of this block to indicate long-term plans to add WaveformExtractor support?

Reply:
Done. Feel free to add something else that captures the current state of affairs.

Reply:
Nah that's good, thanks.